[FC-0059] feat: New fields in library details API view

ChrisChV opened the pull request:
<!--

Note: Please refer to the Support Development Guidelines on the wiki page to consider backporting to active releases:
https://openedx.atlassian.net/wiki/spaces/COMM/pages/4248436737/Support+Guidelines+for+active+releases

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly readable.
If the linked information must be private (because it contains secrets), clearly label the link as private.

-->

## Description

1. This adds the following fields in the library details API view:
- last_draft_created
- last_draft_created_by
- published_by
- created
- updated

2. Delete `collection_uuid` from serializers

## Supporting information

- This change is used in https://github.com/openedx/frontend-app-course-authoring/pull/1138
- Github Issue: https://github.com/openedx/frontend-app-course-authoring/issues/1041
- Internal ticket: [FAL-3760](https://tasks.opencraft.com/browse/FAL-3760).

## Testing instructions

Use this branch and follow the testing instructions on https://github.com/openedx/frontend-app-course-authoring/pull/1138

ChrisChV committed:
feat: Add created and updated field to library details api response

openedx-webhooks commented:
Thanks for the pull request, @ChrisChV!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

ChrisChV committed:
feat: New updates to allow return new fields of library in API view

- create_library_block updated to allow save user that creates the block
- publish_changes updated to allow save user that publish the changes.
- published_by, last_draft_created and last_draft_created_by added to get_library response

ChrisChV committed:
Merge branch 'master' into chris/FAL-3760-update-content-libraries-api

ChrisChV committed:
style: Lint on code

ChrisChV committed:
Merge branch 'master' into chris/FAL-3760-update-content-libraries-api

ChrisChV committed:
Merge branch 'master' into chris/FAL-3760-update-content-libraries-api

ChrisChV committed:
refactor: Use only one query to get has_unpublished_changes in library details api

ChrisChV committed:
refactor: delete collection_uuid from serializers

ChrisChV committed:
Merge branch 'master' into chris/FAL-3760-update-content-libraries-api

rpenido commented on uwsgi.ini:
We should remove this.

rpenido reviewed (approved):
LGTM :+1:
Thank you for your work @ChrisChV!

- [x] I tested this using the instructions from: https://github.com/openedx/frontend-app-course-authoring/pull/1138
- [x] I read through the code
- [x] ~I checked for accessibility issues~
- [x] Includes documentation

bradenmacdonald reviewed (approved):
LGTM, conditional approval if you remove `uwsgi.ini`

ChrisChV committed:
fix: uwsgi.ini deleted

ChrisChV committed:
Merge branch 'master' into chris/FAL-3760-update-content-libraries-api

ChrisChV commented:
@bradenmacdonald I deleted `uwsgi.ini`: https://github.com/openedx/edx-platform/pull/35091/commits/ba201ad5f7c3849288be774b4268648295b11b4d

openedx-webhooks commented:
@ChrisChV 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/35091&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2024-07-05+09:25&entry.1917517419=2024-08-01+18:50&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
